### PR TITLE
ENG-112 Implement Sequence ID

### LINKED ISSUE #2741: Partition (sequence) ID support
Github issue corresponding to https://jira.numenta.com/browse/RES-112

We want to be able to pass the sequence ID info throughout the network; this is useful for debugging purposes. The [partitionID in the KNNClassifier|https://github.com/numenta/nupic/blob/master/src/nupic/algorithms/KNNClassifier.py#L359] is supposed to do this, but it's not clear this works. So the task is to first test to see if this is functioning properly. Then we need to pass the sequence (partition) ID number throughout the network. The LanguageSensor already has an output 

["sequenceIdOut"|https://github.com/numenta/nupic.research/blob/master/htmresearch/regions/LanguageSensor.py#L101], but this needs to be linked to the subsequent regions in the network. This requires (i) that these regions (SP, TM, TP, KNN, and CLA) have inputs for "sequenceIdIn", and (ii) that the links are created by the network factory...
i. add to the the spec:
          sequenceIdIn=dict(
            description="Sequence ID",
            dataType='UInt64',
            count=1,
            required=False,
            regionLevel=True,
            isDefaultInput=False,
            requireSplitterMap=False)
ii. add the following in the network factory's linking method:
  network.link(sensorRegionName, currentRegionName, "UniformLink", "",
               srcOutput="sequenceIdOut", destInput="sequenceIdIn")

### PR DESCRIPTION
**Pull request by SaganBolliger**
https://jira.numenta.com/browse/RES-112 (fixes #2741)

To be merged concurrently with numenta/nupic.research#346

@BoltzmannBrain Ready for review

### CODE REVIEW

File: src/nupic/regions/KNNClassifierRegion.py [hunk 1]
```diff
@@ -844,6 +854,9 @@ def compute(self, inputs, outputs):
     TODO: confusion matrix does not support multi-label classification
     """
 
+    import pprint
+    pprint.pprint(inputs)
```
Comment (BoltzmannBrain) [inline src/nupic/regions/KNNClassifierRegion.py]: Did you forget to remove this?

File: src/nupic/research/spatial_pooler.py [hunk 2]
```diff
@@ -110,25 +110,25 @@ class SpatialPooler(object):
...
-               synPermConnected=0.10,
-               minPctOverlapDutyCycle=0.001,
-               minPctActiveDutyCycle=0.001,
-               dutyCyclePeriod=1000,
-               maxBoost=10.0,
-               seed=-1,
-               spVerbosity=0,
-               wrapAround=True
-               ):
+              inputDimensions=(128, 128),
```
Comment (BoltzmannBrain) [inline src/nupic/research/spatial_pooler.py]: We don't want to change the default SP params...

**Comment by BoltzmannBrain** [comment]
@SaganBolliger needs some cleanup

**Comment by SaganBolliger** [comment]
@BoltzmannBrain Sorry about this--that was sloppy of me to leave in the print statements and custom parameters.

**Comment by BoltzmannBrain** [comment]
LGTM
@scottpurdy please take a look and merge if okay. Do you think we need unit tests to cover this?

**Pull request closed without merging**